pkg/api/handlers/mcpcatalogs.go

```diff
@@ -28,15 +29,17 @@ type MCPCatalogHandler struct {
 	sessionManager     *mcp.SessionManager
 	oauthChecker       MCPOAuthChecker
 	gatewayClient      *gclient.Client
+	storageClient      kclient.Client
 }
 
-func NewMCPCatalogHandler(defaultCatalogPath string, serverURL string, sessionManager *mcp.SessionManager, oauthChecker MCPOAuthChecker, gatewayClient *gclient.Client) *MCPCatalogHandler {
+func NewMCPCatalogHandler(defaultCatalogPath string, serverURL string, sessionManager *mcp.SessionManager, oauthChecker MCPOAuthChecker, gatewayClient *gclient.Client, storageClient kclient.Client) *MCPCatalogHandler {
 	return &MCPCatalogHandler{
 		defaultCatalogPath: defaultCatalogPath,
 		serverURL:          serverURL,
 		sessionManager:     sessionManager,
 		oauthChecker:       oauthChecker,
 		gatewayClient:      gatewayClient,
+		storageClient:      storageClient,
```

Comment:
I believe you can use the `Storage` field on the `api.Context` passed to your handler methods instead of adding a new field to the handler type.

Reply:
Thanks, just pushed the fix